### PR TITLE
Tomhaile/ch9532/standardize flash scripts parameters

### PR DESCRIPTION
**Pull request by bthaile**
Use options parser throughout flash scripts. Beefed up help.

### CODE REVIEW

File: scripts/flash/dispute-contribute.js [hunk 1]
```diff
@@ -91,61 +97,30 @@ function disputeContributeInternal(augur, marketId, outcome, amount, disputerAut
...
+      }
+      amount = balances.reputation;
       console.log(chalk.yellow.dim("amount"), amount);
-      disputeContributeInternal(augur, marketId, outcomeId, amount, userAuth, invalid, auth, callback);
-    }
+      disputeContributeInternal(augur, marketId, outcome, amount, userAuth, invalid, auth, callback);
+    });
+  } else {
+    console.log(chalk.yellow.dim("amount"), amount);
+    disputeContributeInternal(augur, marketId, outcome, amount, userAuth, invalid, auth, callback);
```
Comment (adrake33) [inline scripts/flash/dispute-contribute.js]: Recommend moving these 2 lines outside the if/else block and removing the else block.

File: scripts/flash/fork.js [hunk 2]
```diff
@@ -7,14 +7,24 @@ var getTime = require("./get-timestamp");
...
  * Move time to Market end time and do go to a fork
  */
-function forkInternal(augur, marketId, userAuth, auth, callback) {
+function fork(augur, args, auth, callback) {
+  if (args === "help" || args.opt.help) {
+    help();
+    return callback(null);
+  }
+  var marketId = args.opt.marketId;
+  var userAuth = auth;
```
Comment (adrake33) [inline scripts/flash/fork.js]: `userAuth` seems redundant. I think you could just use auth everywhere.

File: scripts/flash/designated-report.js [hunk 3]
```diff
@@ -9,13 +9,25 @@ var setTimestamp = require("./set-timestamp");
...
  */
-function designateReportInternal(augur, marketId, outcome, userAuth, invalid, auth, callback) {
+function designateReport(augur, args, auth, callback) {
+  if (args === "help" || args.opt.help) {
+    help();
+    return callback(null);
+  }
   var amount = 10000;
+  var marketId = args.opt.marketId;
+  var userAuth = auth;
```
Comment (adrake33) [inline scripts/flash/designated-report.js]: It seems odd to set `userAuth` = `auth` here when later in the code there's this line:

```
if (userAuth.address !== market.designatedReporter && auth.address === market.designatedReporter) {
        userAuth = auth;
}
```

File: scripts/flash/index.js [hunk 4]
```diff
@@ -30,55 +31,101 @@ var NETWORKS = ["aura", "clique", "environment", "rinkeby", "ropsten"];
...
+    opts: {
+      help: {flag: true, short: "h", help: "This help, dispute this market all the way to fork" },
+      marketId: { required: true, short: "m", help: "Required market id" },
+    },
   },
   "approval": {
     method: approval,
-    opts: null,
+    opts: {
+      help: {flag: true, short: "h", help: "This help" },
```
Comment (adrake33) [inline scripts/flash/index.js]: Why does `help` always start with "This help"?  I think this one and the next one should have further descriptions.

File: scripts/flash/index.js [hunk 5]
```diff
@@ -232,38 +255,31 @@ if (require.main === module) {
...
-    help();
-    process.exit();
+
+  if (args.opt.help) {
+    if (!method) {
+      help();
+      process.exit();
+    } else {
+      options.help(method.opts);
+      process.exit();
```
Comment (adrake33) [inline scripts/flash/index.js]: This line can be moved out of here below the if/else block.

File: scripts/flash/initial-report.js [hunk 6]
```diff
@@ -7,13 +7,25 @@ var getTime = require("./get-timestamp");
...
  */
-function initialReportInternal(augur, marketId, outcome, userAuth, invalid, auth, callback) {
+function initialReport(augur, args, auth, callback) {
+  if (args === "help" || args.opt.help) {
+    help();
+    return callback(null);
+  }
+  var marketId = args.opt.marketId;
+  var outcome = args.opt.outcome;
+  var userAuth = auth;
```
Comment (adrake33) [inline scripts/flash/initial-report.js]: Recommend just using `auth` here as well.

File: scripts/flash/push-time.js [hunk 7]
```diff
@@ -3,80 +3,16 @@
...
-            modTimeBy = 1; // seconds
-          }
-          var totalMovement = modTimeBy * digit;
-          var newTimestamp = subtraction ? timestamp - totalMovement : timestamp + totalMovement;
-          setTimestamp(augur, newTimestamp, timeResult.timeAddress, auth, callback);
-        }
-      });
-    });
-  }
+function pushTime(augur, args, auth, callback) {
```
Comment (adrake33) [inline scripts/flash/push-time.js]: Are you passing in `augur`, `args`, & `auth` here for consistency? They're not used in this function.

File: scripts/flash/show-initial-reporter.js [hunk 8]
```diff
@@ -5,10 +5,20 @@
...
+function help() {
+  console.log(chalk.red("Show initial reporter account for market"));
+  console.log(chalk.red("Displays all 0s if market hasn't been reported on"));
+}
+
 /**
  * Move time to Market end time and do initial report
  */
-function showInitialReporterInternal(augur, marketId, auth, callback) {
+function showInitialReporter(augur, args, auth, callback) {
```
Comment (adrake33) [inline scripts/flash/show-initial-reporter.js]: `auth` param doesn't get used here.  I didn't check every function in this PR for unused params, but if this is something that should be corrected, I recommend double-checking the other functions in this PR as well.

File: scripts/flash/push-time.js (hunk 7 above)
Comment (bthaile) [reply scripts/flash/push-time.js]: yes, that is the pattern. every command gets passed these options.

File: scripts/flash/show-initial-reporter.js (hunk 8 above)
Comment (bthaile) [reply scripts/flash/show-initial-reporter.js]: Every command gets passed these parameters/options. It's like an interface if we were using typescript.

**Pull request merged**